dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/TasksRunner.java

```diff
@@ -152,21 +155,26 @@ private <T> T runTask(Task<T> task) throws MetadataDumperUsageException {
       // method instead of this runTask() method.
       if (!task.handleException(e)) LOG.warn("Task failed: " + task + ": " + e, e);
       state.setTaskException(task, TaskState.FAILED, e);
-      try {
-        OutputHandle sink = context.newOutputFileHandle(task.getTargetPath() + ".exception.txt");
-        sink.asCharSink(StandardCharsets.UTF_8)
-            .writeLines(
-                Arrays.asList(
-                    task.toString(),
-                    "******************************",
-                    String.valueOf(new DumperDiagnosticQuery(e).call())));
-      } catch (Exception f) {
-        LOG.warn("Exception-recorder failed:  " + f, f);
-      }
+      createExceptionFile(task, e);
     }
     return null;
   }
 
+  private void createExceptionFile(Task<?> task, Exception e) {
+    try {
+      ImmutableList.Builder<String> lines = ImmutableList.builder();
+      lines.add(task.toString()).add("******************************");
```

Comment:
Do we need the stars if `runDiagnosticQueryError` is `false`?

Reply:
We don't need the stars. However, I wanted to omit the diagnostic query only, leaving everything else as is, so as to affect the dumper execution as little as possible.